docs/develop/gettingstarted.rst

```diff
@@ -0,0 +1,300 @@
+   or any other sources on the internet.
+
+Get the code
+----------------
+
+It is recommended that you maintain a fork of the MicroPython repository for your development purposes.
+The process of obtaining the source code includes the following:
+
+| Fork the repository https://github.com/micropython/micropython.git
+| You will now have a fork at <https://github.com/<your-user-name>/micropython.git>.
+| Clone the repository. You may clone either your fork or the main MicroPython repository.
+   Use one of the commands below:
+
+.. code-block:: console
+
+   $ git clone https://github.com/micropython/micropython.git 
+   $ git clone https://github.com/<your-user-name>/micropython.git
+
+Depending on what you cloned, `configure the remote 
+repositories <https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes>`_ to be able to
+collaborate on the MicroPython project.
+
+Configure upstream if you cloned your fork.
+
+.. code-block:: console
+
+   $ cd micropython
+   $ git remote add upstream https://github.com/micropython/micropython.git
+
+It is common to configure ``upstream`` and ``origin`` on a forked repository
+to assist with sharing code changes. You can maintain your own mapping but
+it is recommended that ``origin`` maps to your fork and ``upstream`` to the main
+MicroPython repository:
+
+.. code-block:: console
+
+   $ cd micropython
+   $ git remote add upstream https://github.com/micropython/micropython.git
+   $ git remote set-url origin https://github.com/<your-user-name>/micropython.git
+
+After the above configuration, your setup should be similar to this:
+
+.. code-block:: console
+   
+   $ git remote -v
+   origin	https://github.com/<your-user-name>/micropython.git (fetch)
+   origin	https://github.com/<your-user-name>/micropython.git (push)
+   upstream	https://github.com/micropython/micropython.git (fetch)
+   upstream	https://github.com/micropython/micropython.git (push)
+
+You should now have a copy of the source code. By default you are pointing
+to the master branch. To prepare for further development, it is recommended
+to work on a development branch.
+
+.. code-block:: console
+
+    $ git checkout -b dev-branch
+
+You can give it any name. You will have to compile MicroPython whenever you change 
+to a different branch.
+
+Compile and build the Code
+--------------------------
+
+For development, you should compile MicroPython for a specific platform which corresponds
+to a port. Start by installing the required dependencies. Then build the MicroPython 
+cross-compiler before you can successfully compile and build.
+
+Required dependencies
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Before anything, make sure you have a working environment for ``gcc`` and ``GNU make``.
+Ubuntu 20.04 has been used for the example below but other unixes ought to work with
+little modification:
+
+.. code-block:: console
+
+   $ gcc --version
```

Comment:
Imo this should either go after the instructions to install it, or you should explain this more in detail - I assume (not sure though) what you mean her is something like 'if you can run `gcc --version` and get output like below you might already have installed what is needed [for the unix port]'. However it's not because gcc runs that one can build MicroPython.

Reply:
I might move this back as a note under the unix port, thoughts?

Reply:
Not sure, `gcc` is still needed to build mpy-cross so it is not specific to the unix port only